store/tikv/latch/latch.go

```diff
@@ -219,24 +207,28 @@ func (latches *Latches) popFromWaitingQueue(slotID int) (front *Lock, hasMoreWai
 	return
 }
 
-func (latches *Latches) acquireSlot(slotID int, lock *Lock) acquireResult {
+func (latches *Latches) acquireSlot(slotID int, lock *Lock) {
 	latch := &latches.slots[slotID]
 	latch.Lock()
 	defer latch.Unlock()
-	if latch.maxCommitTS > lock.startTS {
-		return acquireStale
-	}
-	// Empty latch.
-	if !latch.occupied() {
+	if !latch.occupied() || latch.waitingQueueHead == lock.startTS {
 		latch.waitingQueueHead = lock.startTS
+		lock.acquiredCount++
+	}
+
+	if latch.maxCommitTS > lock.startTS {
```

Comment:
Buggy here. Answer me a question: 
If a lock try to acquireSlot and get stale, should it hold the last slot or not (acquiredCount)?

In the code here, if it's called by a txn, it doesn't hold the last slot (acquiredCount = N)
if the code is called by scheduler, it runs to line 216 and also here, hold the slot (acquiredCount = N+1)

I really suggest split those two case into different functions.